modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -1202,7 +1213,92 @@ private void execLoadOpcodes(StackFrame sf, int opcode, int[] operands) {
                 sf.refRegs[j] = structureType.getRefField(fieldIndex);
                 break;
 
-            case InstructionCodes.MAPLOAD:
+            case InstructionCodes.IMAPLOAD:
+                i = operands[0];
+                j = operands[1];
+                k = operands[2];
+                bMap = (BMap<String, BRefType>) sf.refRegs[i];
+                if (bMap == null) {
+                    handleNullRefError();
+                    break;
+                }
+
+                bRefType = bMap.get(sf.stringRegs[j]);
+                if (bRefType != null) {
+                    sf.longRegs[k] = ((BInteger) bRefType).intValue();
+                } else {
+                    sf.longRegs[k] = 0;
+                }
+                break;
+            case InstructionCodes.FMAPLOAD:
+                i = operands[0];
+                j = operands[1];
+                k = operands[2];
+                bMap = (BMap<String, BRefType>) sf.refRegs[i];
+                if (bMap == null) {
+                    handleNullRefError();
+                    break;
+                }
+
+                bRefType = bMap.get(sf.stringRegs[j]);
+                if (bRefType != null) {
+                    sf.doubleRegs[k] = ((BFloat) bRefType).floatValue();
+                } else {
+                    sf.doubleRegs[k] = 0;
+                }
+                break;
+            case InstructionCodes.SMAPLOAD:
+                i = operands[0];
+                j = operands[1];
+                k = operands[2];
+                bMap = (BMap<String, BRefType>) sf.refRegs[i];
+                if (bMap == null) {
+                    handleNullRefError();
+                    break;
+                }
+
+                bRefType = bMap.get(sf.stringRegs[j]);
+                if (bRefType != null) {
+                    sf.stringRegs[k] = ((BString) bRefType).stringValue();
+                } else {
+                    sf.stringRegs[k] = STRING_NULL_VALUE;
+                }
+                break;
+            case InstructionCodes.BMAPLOAD:
+                i = operands[0];
+                j = operands[1];
+                k = operands[2];
+                bMap = (BMap<String, BRefType>) sf.refRegs[i];
+                if (bMap == null) {
+                    handleNullRefError();
+                    break;
+                }
+
+                bRefType = bMap.get(sf.stringRegs[j]);
+                if (bRefType != null) {
+                    sf.intRegs[k] = ((BBoolean) bRefType).booleanValue() == BBoolean.TRUE.booleanValue() ? 1 : 0;
```

Comment:
This can be simplified into` ((BBoolean) bRefType).booleanValue() ? 1 : 0;`

Reply:
+1